lib/src/builder/codegen/util.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/src/component_declaration/annotations.dart' as annotations;
+import 'package:source_span/source_span.dart';
+
+import '../parsing.dart';
+import 'names.dart';
+
+abstract class Generator {
```

Comment:
I'm a little wary of overloading the `Generator` class in a build context (`Generator` class also exists in `source_gen`). Probably not a huge deal, so consider this a very optional #nit.

Reply:
Good point; I can definitely make this not collide, and rename it to `BoilerplateDeclarationGenerator` or something